HIP/hip-1208.md

```diff
@@ -0,0 +1,78 @@
+---
+hip: 1208
+title: Native integration of Hedera Smart Contract Service (HSCS) with Hedera Consensus Service (HCS) topics
+author: Walter Hernandez <walter.hernandez.18@ucl.ac.uk>, Juan Ignacio Ibañez <j.ibanez@ucl.ac.uk>, Paolo Tasca <p.tasca@exp.science>, Nikhil Vadgama <nikhil.vadgama@exp.science>, Jiahua Xu <jiahua.xu@ucl.ac.uk>
+working-group: 
+requested-by:  Walter Hernandez <walter.hernandez.18@ucl.ac.uk>, Juan Ignacio Ibañez <j.ibanez@ucl.ac.uk>, Paolo Tasca <p.tasca@exp.science>, Nikhil Vadgama <nikhil.vadgama@exp.science>, Jiahua Xu <jiahua.xu@ucl.ac.uk>
+type: Standards Track
+category: Service
+needs-hedera-review: Yes
+hedera-review-date: 
+hedera-approval-status:
+needs-hiero-approval: Yes
+status: Draft
+created: 2025-06-02
+discussions-to: https://github.com/hiero-ledger/hiero-improvement-proposals/discussions/1207
+updated: 2025-06-02
+requires:
+replaces:
+superseded-by:
+---
+
+## Abstract
+This proposal enables direct intraoperability between Hedera Smart Contracts and topics created using the Hedera Consensus Service (HCS). Smart contracts will be able to read and write to HCS topics through native integration. This allows for real-time event processing, decentralized messaging, and cross-application coordination without external intermediaries like Oracles.
+
+## Motivation
+Developers will be able to leverage HCS topics for secure, verifiable messaging and event tracking directly within smart contract logic, opening up use cases such as decentralized coordination, event-driven automation, and consensus-based contract execution.
+
+## Rationale
+Allowing smart contracts to interact natively with HCS topics enables applications to maintain tamper-proof records, trigger contract execution based on consensus-driven data, and create verifiable off-chain attestations. For example:
+- A smart contract can listen to HCS topics and execute predefined actions based on received messages.
+- HCS topics can serve as permissioned registries, where contracts validate participants' eligibility before executing transactions.
+- Multiple smart contracts on Hedera and external applications (off-graph) can read and write from the same topics, enabling many-to-many communication patterns. For example:
+  - External applications can post messages to HCS topics that smart contracts can react to.
+  - Smart contracts can emit events to topics that external applications listen to.
+- The Hedera mainnet provides the trusted consensus layer that ensures all participants within the same topics see the same sequence of events.
+
+## User stories
+- As a developer, I want my smart contract to read data from HCS topics to trigger predefined on-chain actions in a verifiable manner.
```

Comment:
What would the specification look like for reading topics? How do you envision this working?

Reply:
The HIP has been updated to focus on the writing to topics, leaving the reading from topics as needing another completely different HIP